### PR TITLE
UI tests for provider/client

### PR DESCRIPTION
**Pull request by ebondare**
Add UI tests for provider/client setup

### CODE REVIEW

File: ocs_ci/ocs/ui/page_objects/page_navigator.py [hunk 1]
```diff
@@ -338,14 +338,26 @@ def navigate_block_pool_page(self):
...
             self.nav_odf_default_page()
             .nav_storage_systems_tab()
             .nav_storagecluster_storagesystem_details()
         )
         storage_system_details.nav_ceph_blockpool()
-        logger.info("Now at Block pool page")
+        logger.info("Now at Block Pools page")
+
+    def navigate_client_page(self):
+        """
```
Comment (DanielOsypenko) [inline ocs_ci/ocs/ui/page_objects/page_navigator.py:351]: We need a new class for such functions. Page Navigator is the side-bar in the left side of the console. Can we create a class StorageCLients for the functionality available in the red square?

image

Comment (ebondare) [reply ocs_ci/ocs/ui/page_objects/page_navigator.py:351]: Yes, that's what I'm doing in client_ui.py file. I can rename it to storageclients if that's what you are suggesting.

File: ocs_ci/ocs/ui/client_ui.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    User Interface Selenium for Clients page
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def generate_new_token(self):
+        """
+        Create a new client onboarding token in the UI
```
[the remark this answers is not in the document]
Comment (DanielOsypenko) [reply ocs_ci/ocs/ui/client_ui.py:23]: I would call the class StorageClientsOnProvider and I would suggest to have one module provider_client_ui.py for the class StorageClientsOnProvider and for the coming soon ClientDashboard class. 

We can separate them later if see that it is not logically or it is getting huge.

File: ocs_ci/ocs/ui/page_objects/page_navigator.py (hunk 1 above)
Comment (DanielOsypenko) [reply ocs_ci/ocs/ui/page_objects/page_navigator.py:351]: Yes please, def navigate_storage_clients_page should be good, wdyt?

File: ocs_ci/ocs/ui/page_objects/page_navigator.py [hunk 3]
```diff
@@ -338,14 +338,26 @@ def navigate_block_pool_page(self):
...
+        Navigate to clients page (for Provider/Client setup)
```
Comment (DanielOsypenko) [inline ocs_ci/ocs/ui/page_objects/page_navigator.py:352]: This method should return instance of the class StorageClients, similarly to other navigate methods

File: ocs_ci/ocs/ui/client_ui.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ebondare) [reply ocs_ci/ocs/ui/client_ui.py:23]: OK, thanks, I'll make the changes

File: ocs_ci/ocs/resources/storageconsumer.py [hunk 4]
```diff
@@ -55,6 +56,34 @@ def get_ocs_version(self):
...
+            string: heartbeat in format "2024-01-24T10:10:01Z"
+
+        """
+        return self.ocp.get(resource_name=self.name).get("status").get("lastHeartbeat")
+
+    def is_heartbeat_ok(self):
+        """
+        Checks if last heartbeat was less than 5 minutes ago
+
+        Returns:datetime.datetime.now()
```
Comment (dahorak) [inline ocs_ci/ocs/resources/storageconsumer.py:73]:
```suggestion
        Returns:
```

File: ocs_ci/ocs/ui/provider_client_ui.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                clients_with_heartbeat += 1
+        assert clients_with_heartbeat == int(connected_clients), (
+            f"Number of connected clients on the dashboard: {connected_clients}",
+            f"Number of clients with recent heartbeat: {clients_with_heartbeat}",
+        )
+
+    def get_token_description(self):
+        """
+        Verify that token window has description how to use it
+        """
```
Comment (dahorak) [inline ocs_ci/ocs/ui/provider_client_ui.py:103]: Please add `Returns:` section to the doc string.

File: ocs_ci/ocs/resources/storageconsumer.py (hunk 4 above)
Comment (dahorak) [inline ocs_ci/ocs/resources/storageconsumer.py:73]:
```suggestion
        Returns:
```

File: ocs_ci/ocs/ui/provider_client_ui.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        consumer_names = get_all_storageconsumer_names()
+        for consumer_name in consumer_names:
+            consumers_obj = ocp.OCP(
+                kind=constants.STORAGECONSUMER,
+                namespace=config.cluster_ctx.ENV_DATA["cluster_namespace"],
+                resource_name=consumer_name,
+            )
+            client_name = consumers_obj.get("status").get("client").get("name")
+            logger.info(f"Search for {client_name} client")
+            self.do_send_keys(self.validation["search_client"], text=client_name)
```
Comment (DanielOsypenko) [inline ocs_ci/ocs/ui/provider_client_ui.py:52]: can't find here that we clean-up input value.
If no clean-up we will get next client_name merged with previous

**Pull request closed without merging**